### PR TITLE
Add rubocop

### PR DESCRIPTION
**Pull request by ehelms**
Copied from katello-installer setup

### CODE REVIEW

File: config/foreman.migrations/20160420224417_puppet_autosign.rb
Comment (ekohl) [inline config/foreman.migrations/20160420224417_puppet_autosign.rb]: Looks like this breaks: `undefined method `to_bool' for nil:NilClass`

File: Rakefile
Comment (ekohl) [inline Rakefile]: Personally I'm a fan of trailing commas because it makes diffs smaller when you add another line

File: .rubocop_todo.yml
Comment (ekohl) [inline .rubocop_todo.yml]: Could this be updated?

**Review by ekohl — CHANGES_REQUESTED** [review]

File: Rakefile
Comment (ehelms) [reply Rakefile]: Which I get in puppet land, but in Ruby this is generally not the case to be seen.

**Comment by ehelms** [comment]
Updated

File: config/foreman.migrations/20160420224417_puppet_autosign.rb
Comment (ekohl) [inline config/foreman.migrations/20160420224417_puppet_autosign.rb]: Would `(current_autosign.nil? || current_autosign.to_bool == current_autosign)` be equivalent?

**Review by ekohl — COMMENTED** [review]
I've added a commit that gets rid of `.rubocop_todo.yml`. Please have a look if you think it improves the situation or if we need to disable more cops.

**Comment by ehelms** [comment]
@ekohl The only argument against adding our commit *now* is that you will need to do the same to katello-installer. The route I was going was to bring foreman-installer to the same level as katello-installer to transition the code in and then make updates across the entire codebase at once.

**Comment by ekohl** [comment]
That's why I was preferring to add rubocop after the merge: to avoid a todo file in the history.

**Comment by ehelms** [comment]
@ekohl Up to you, not sure what a todo file in our history is hurting.

**Review by chris1984 — APPROVED** [review]

**Review by chris1984 — APPROVED** [review]
LGTM im indifferent to the todo, im always trying to fix things and clear the todo, other than a rebase to fix the conflicts i would say ok.

**Comment by mmoll** [comment]
@ehelms needs a rebase

**Comment by ekohl** [comment]
I have a WIP branch that covers most. Will get back to that on Monday.

**Comment by ehelms** [comment]
@ekohl What did you want to do with respect to rubocop adding?

**Comment by ehelms** [comment]
I've gone and updated this to the HEAD of develop and started the base configuration of cops to that used in Foreman core.

File: Rakefile
Comment (ekohl) [reply Rakefile]: We actually apply the trailing comma in Foreman as well.

File: katello/hooks/boot/01-kafo-hook-extensions.rb
Comment (ekohl) [inline katello/hooks/boot/01-kafo-hook-extensions.rb]: I must say I really dislike the cop that enforces this style. Can we disable it?

File: katello/hooks/pre/30-upgrade.rb
Comment (ekohl) [inline katello/hooks/pre/30-upgrade.rb:20]: There's also a typo in `enconding` here.

File: katello/hooks/pre/30-upgrade.rb
Comment (ekohl) [inline katello/hooks/pre/30-upgrade.rb:45]:
```suggestion
  FileUtils.mkdir_p(STEP_DIRECTORY)
```

**Review by ekohl — CHANGES_REQUESTED** [review]
I'd like to enforce trailing commas

File: katello/hooks/boot/01-kafo-hook-extensions.rb
Comment (ehelms) [reply katello/hooks/boot/01-kafo-hook-extensions.rb]: We can if you like. I :heart: guard clauses.

Comment (ekohl) [reply katello/hooks/boot/01-kafo-hook-extensions.rb]: They can be good and useful but not always. The cop that enforces it is too strict IMHO. I've had cases where I preferred the explicit long version because I thought it was easier to read. IMHO this is also one of them.

**Comment by ehelms** [comment]
Dropped the guard clauses but I'd like to hold this till my post install hook merge is done.

**Comment by ehelms** [comment]
Rebased and updated

File: .gitignore
Comment (evgeni) [inline .gitignore:34]: Where does this folder come from? And should it also be in the `AllCops/Exclude` list if that's where `bundle install` installs stuff.

Comment (ehelms) [reply .gitignore:34]: Rubocop seems to ignore hidden directories by default.

**Comment by mmoll** [comment]
@ehelms please rebase :)

**Review by mmoll — APPROVED** [review]

**Comment by evgeni** [comment]
@ehelms needs another rebase :(

**Comment by mmoll** [comment]
```yaml
Style/TrailingCommaInArrayLiteral:
  EnforcedStyleForMultiline: comma

Style/TrailingCommaInHashLiteral:
  EnforcedStyleForMultiline: comma
```

Should get added. This is, what we have in core.

**Comment by ehelms** [comment]
> ```yaml
> Style/TrailingCommaInArrayLiteral:
>   EnforcedStyleForMultiline: comma
> 
> Style/TrailingCommaInHashLiteral:
>   EnforcedStyleForMultiline: comma
> ```
> 
> Should get added. This is, what we have in core.

We have to pin to an older version of Rubocop due to Ruby 2.0 on EL7. That means this cops aren't available I think:

```
Warning: unrecognized cop Style/TrailingCommaInArrayLiteral found in /home/ehelms/workspace/upstream/foreman-installer/.rubocop.yml
Warning: unrecognized cop Style/TrailingCommaInHashLiteral found in /home/ehelms/workspace/upstream/foreman-installer/.rubocop.yml
```

**Comment by mmoll** [comment]
`Style/TrailingCommaInLiteral` only then, perhaps? :)

**Comment by mmoll** [comment]
Pushed with `Style/TrailingCommaInLiteral`

File: Puppetfile
Comment (ekohl) [inline Puppetfile]: This is an odd move.

**Review by ekohl — APPROVED** [review]

File: Puppetfile
Comment (mmoll) [reply Puppetfile]: removed.

**Comment by ekohl** [comment]
I introduced merge conflicts via #452.

**Comment by ekohl** [comment]
Also looks like Rubocop doesn't like the spacing in the Puppetfile.

**Comment by mmoll** [comment]
Excluded `Puppetfile` and fixed everything.

**Pull request merged**
